react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      visibleIndex: null
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    const target = event.target.id;
+    for (let i = 0; i < this.props.data.length; i++) {
+      if (target === this.props.data[i].title) {
+        this.setState({ visibleIndex: i });
+      }
+    }
+  }
```

Comment:
Instead of looping to find the correct index, you can have the `index` directly passed into the `handleClick` method:
```JAVASCRIPT
handleClick(index) {
  this.setState({ visibleIndex: index });
}
```